### PR TITLE
add IPFS protocol links support

### PR DESCRIPTION
**Pull request by ghost**
Make native IPFS links clickable. Construction identical to the 'http' case. Was tested successfully.

### CODE REVIEW

File: js/twister_formatpost.js [hunk 1]
```diff
@@ -698,6 +698,18 @@ function htmlFormatMsg(msg, opt) {
             msg = msgAddHtmlEntity(msg, i, i + str.length,
                 newHtmlEntityLink(twister.tmpl.linkShortened, str, str));
             i = msg.i;
+        } else if (msg.str.slice(i, i + 4).toLowerCase() === 'ipfs' &&
+                   msg.str.slice(i + 4, i + 7) === '://' &&
```
Comment (slr) [inline js/twister_formatpost.js]: no need to separate `ipfs` and `://` here

Comment (slr) [reply js/twister_formatpost.js]:
```suggestion
        } else if (msg.str.slice(i, i + 7).toLowerCase() === 'ipfs://'
```

File: js/twister_formatpost.js [hunk 2]
```diff
@@ -698,6 +698,18 @@ function htmlFormatMsg(msg, opt) {
...
+                   stopCharsRight.indexOf(msg.str[i + 7]) === -1) {
+           j = getSubStrEnd(msg.str, i + 7, stopCharsRight, false, stopCharsTrailingUrl);
+           if (j > i + 6) {
+               str = msg.str.slice(i, j + 1);
+               msg = msgAddHtmlEntity(msg, i, i + str.length,
+                   newHtmlEntityLink(_htmlFormatMsgLinkTemplateExternal,
+                       proxyURL(str), str)
```
Comment (slr) [inline js/twister_formatpost.js]: I guess we do not want to apply `proxyURL()` for IPFS links

Comment (slr) [reply js/twister_formatpost.js]:
```suggestion
                       str, str)
```

**Review by slr — CHANGES_REQUESTED** [review]

**Comment by slr** [comment]
also notice please we do not merge pulls right into `master` branch. read [readme](https://github.com/miguelfreitas/twister-html#contribute) again

**Comment by slr** [comment]
also notice please we do not merge pulls right into `master` branch. read [readme](https://github.com/miguelfreitas/twister-html#contribute) again

**Comment by ghost** [comment]
> I guess we do not want to apply `proxyURL()` for IPFS links

Oops... my mistake.

> To make it easier for us to accept your patches, please follow the conventional GitHub workflow and keep in mind that your pull requests should have blaster branch as both the origin and target.

Hello) thanks for fixes, accepted - will pull updates into the blaster branch next time!

**Comment by D4708** [comment]
As understand it right, I should send the feature proposes to the both of branches or only into the blaster one?
p.s. not sure about playing with ipfs/ipns condition optimization, just copied previous construction in https://github.com/miguelfreitas/twister-html/pull/381/commits/0b66f70ca145151b363d8ad5c7f03ad64dab6344

**Comment by ghost** [comment]
#383

**Comment by slr** [comment]
@D4708 sorry for delay. I wasn't able to test it since I almost don't use my laptop these days - if only we have twister for mobiles!

**Comment by ghost** [comment]
Now I understand how does contribution work. I've tested each commit before pull request, other else mark it as ~untested~. 
Just checked - https://github.com/miguelfreitas/twister-html/pull/383 works fine. 

Thank you for fixes and project maintenance!

**Pull request closed without merging**